internal/provider/resource_eva_employee.go

```diff
@@ -0,0 +1,276 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-go/tftypes"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+
+	"github.com/mad-it/terraform-provider-eva/internal/eva"
+)
+
+type employeeType struct{}
+
+func (t employeeType) GetSchema(ctx context.Context) (tfsdk.Schema, diag.Diagnostics) {
+	return tfsdk.Schema{
+		MarkdownDescription: "Eva employee configuration.",
+
+		Attributes: map[string]tfsdk.Attribute{
+			"id": {
+				MarkdownDescription: "ID of the employee.",
+				Computed:            true,
+				Type:                types.Int64Type,
+				PlanModifiers: tfsdk.AttributePlanModifiers{
+					tfsdk.UseStateForUnknown(),
+				},
+			},
+			"first_name": {
+				Optional: true,
+				Type:     types.StringType,
+			},
+			"last_name": {
+				Optional: true,
+				Type:     types.StringType,
+			},
+			"email_address": {
+				Required: true,
+				Type:     types.StringType,
+			},
+			"password": {
+				Required:  true,
+				Type:      types.StringType,
+				Sensitive: true,
+			},
+			"roles": {
+				MarkdownDescription: "list of scoped functionalities to be attached",
+				Optional:            true,
+				Attributes: tfsdk.ListNestedAttributes(
+					map[string]tfsdk.Attribute{
+						"role_id": {
+							MarkdownDescription: "id of the role",
+							Required:            true,
+							Type:                types.Int64Type,
+						},
+						"user_type": {
+							MarkdownDescription: "functionality scope",
+							Required:            true,
+							Type:                types.Int64Type,
+						},
+						"organization_unit_id": {
+							MarkdownDescription: "id of the organization unit the role applies too.",
+							Optional:            true,
+							Type:                types.Int64Type,
+						},
+					},
+					tfsdk.ListNestedAttributesOptions{},
```

Comment:
`question`: is empty array a valid input ?

Reply:
Thats a good question.... I think on one hand it is, because it might also get rid of unwanted roles. But on the other hand an account without roles is useless....